src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -709,24 +731,22 @@ private class NativeMethods
             internal static extern ErrorCode rd_kafka_position(
                     IntPtr rk, IntPtr partitions);
 
+            // note: producev signature is rd_kafka_producev(rk, ...)
+            // __arglist is an undocumented keyword which deal with it
+            // but doesn't work on standardlib (and undocumented so don't use)
+            // stay simple with one binding for now, but it will be worth benchmarking
+            // the overload with no timestamp, opaque, partition, etc
             [DllImport(DllName, CallingConvention = CallingConvention.Cdecl)]
-            internal static extern IntPtr rd_kafka_produce(
-                    IntPtr rkt,
-                    int partition,
-                    IntPtr msgflags,
-                    IntPtr val, UIntPtr len,
-                    IntPtr key, UIntPtr keylen,
-                    IntPtr msg_opaque);
-
-            [DllImport(DllName, CallingConvention = CallingConvention.Cdecl)]
-            internal static extern IntPtr rd_kafka_producev(
-                IntPtr rkt,
-                int partition,
-                IntPtr msgflags,
-                IntPtr val, UIntPtr len,
-                IntPtr key, UIntPtr keylen,
-                long timestamp,
-                IntPtr msg_opaque);
+            internal static extern ErrorCode rd_kafka_producev(
+                IntPtr rk,
+                ProduceVarTag topicType, string topic,
```

Comment:
i'm worrying about the marshaling of topic name here.
with the old produce method (where topic handles were required instead of topic name) the new topic method has the topic name parameter decorated with the `[MarshalAs(UnmanagedType.LPStr)]` attribute.
But is even that exactly what we want? it results in the topic name being encoded as 'ANSI' - whatever that means - seems a bit dodgy to me.
I'm wondering if ideally what we want is to manually encode as UTF8 into a byte array (and terminate with 0) and pass that.
This would also apply to other methods that marshal strings.
For this PR, decorating with `[MarshalAs(UnmanagedType.LPStr)]` is probably enough (questioning string marshaling is probably a separate PR).